TundraTD/Assets/Scripts/City/Building/ElementalTower.cs

```diff
@@ -0,0 +1,55 @@
+using City.Building.Upgrades;
+using Spells;
+using UnityEngine;
+
+namespace City.Building
+{
+    public class ElementalTower : MonoBehaviour
+    {
+        [SerializeField] private BasicElement towerElement;
+        [SerializeField] private int towerPurchasePrice; 
+        [SerializeField] private ElementalTowerUI elementalTowerUIPrefab;
+        
+        private ElementalTowerUI _elementalTowerUI;
+        private IUpgrade[,] _towerUpgrades;
+        private int _towerUpgradeLevel;
+        
+        public BasicElement TowerElement => towerElement;
```

Comment:
Nit: I guess there could be an auto property that has private set method instead of backing field

Reply:
For example, do it like that:
```suggestion
        public BasicElement TowerElement { get; private set; }
        //... In this class
        TowerElement = <...>;
        //... In other class, for example
        if (TowerElement == ...) ...;
```